### ISSUE TITLE
Same seed producing different results

### ISSUE THREAD

**Issue by nnnfffkkk** [M3RG-IITD/LaMO#3]
Hello there. I am trying to reproduce results by fixing seeds - but it seems like even with torch.manual_seed and torch.cuda.manual_seed, the pipeline always gives different results (at least for the "shared" models). Any clue how this might be solved?

**Comment by Karn3003** [comment]
Hi, yes, using the same seed can still produce different results because not all random sources are fully fixed. You can try the following function to enforce complete seed fixing, but note that it may degrade performance.
```
def set_seed(seed):
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = False
    torch.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)
    np.random.seed(seed)
    random.seed(seed)
```

or following function:

```
def seed_everything(seed: int):
    import random, os
    import numpy as np
    import torch
    
    random.seed(seed)
    os.environ['PYTHONHASHSEED'] = str(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = True
```

**Comment by nnnfffkkk** [comment]
Thank you!
